elixir/apps/domain/lib/domain/actors/group/sync.ex

```diff
@@ -57,6 +58,35 @@ defmodule Domain.Actors.Group.Sync do
     {:ok, {upsert, delete}}
   end
 
+  # Used to make sure we don't accidentally mass delete groups
+  # due to an error in the Identity Provider API (e.g. Google Admin SDK API)
+  defp deletion_circuit_breaker([], _groups_to_delete) do
+    # If there are no groups then there can't be anything to delete
+    :ok
+  end
+
+  defp deletion_circuit_breaker(_groups, []) do
+    :ok
+  end
+
+  defp deletion_circuit_breaker(groups, groups_to_delete) do
+    groups_length = length(groups)
+    deletion_length = length(groups_to_delete)
+
+    delete_percentage = (deletion_length / groups_length * 100) |> round()
+
+    cond do
+      groups_length > 40 and delete_percentage >= 25 ->
```

Comment:
Probably a good idea to log details of when this occurs so we can keep tabs on it.

```suggestion
      groups_length > 40 and delete_percentage >= 25 ->
        Logger.error("Refusing to mass delete groups", groups_length: groups_length, deletion_length: deletion_length, provider_id: provider.id)
```

Reply:
Good call.  I'll need to update the function signature to get the provider in there, but that won't be a problem.